tests/test_information_from_events.py

```diff
@@ -0,0 +1,464 @@
+        ).transact({"from": accounts[A]})
+
+    timestamp = web3.eth.getBlock("latest").timestamp
+    timestamp += (3600 * 24 * 365) * 3
+
+    chain.time_travel(timestamp)
+    chain.mine_block()
+    return currency_network_contract_with_trustlines
+
+
+def calculate_interests(
+    balance: int, delta_time_in_seconds: int, highest_order: int = 15
+) -> int:
+    """
+    This function was taken from the relay server, for a proof of the correctness of this calculation see
+    `relay/tests/unit/network_graph/test_interests.py`: test_interests_calculation_gives_same_result_as_smart_contracts
+    """
+    SECONDS_PER_YEAR = 60 * 60 * 24 * 365
+    INTERESTS_DECIMALS = 2
+    internal_interest_rate = 1000
+
+    intermediate_order = balance
+    interests = 0
+    # Calculate compound interests using taylor approximation
+    for order in range(1, highest_order + 1):
+        intermediate_order = int(
+            intermediate_order
+            * internal_interest_rate
+            * delta_time_in_seconds
+            / (SECONDS_PER_YEAR * 100 * 10 ** INTERESTS_DECIMALS * order)
+        )
+
+        if intermediate_order == 0:
+            break
+        interests += intermediate_order
+
+    return interests
+
+
+def get_interests_for_trustline(currency_network_contract, a, b):
+    """Get all balance changes of a trustline because of interests"""
+    balance_updates = get_balance_updates_for_trustline(currency_network_contract, a, b)
+
+    timestamps = [
+        currency_network_contract.web3.eth.getBlock(balance_update[0][0])["timestamp"]
+        for balance_update in balance_updates
+    ]
+
+    return [
+        calculate_interests(balance_update[1], post_time - pre_time)
+        for (balance_update, pre_time, post_time) in zip(
+            balance_updates[:-1], timestamps[:-1], timestamps[1:]
+        )
+    ]
+
+
+def get_balance_updates_for_trustline(currency_network_contract, a, b):
+    """Get all balance changes of a trustline. Returns a list of (event_key, balance_change)"""
+    balance_update_events = currency_network_contract.events.BalanceUpdate().getLogs(
+        fromBlock=0, argument_filters={"_from": a, "_to": b}
+    )
+    reverse_balance_update_events = currency_network_contract.events.BalanceUpdate().getLogs(
+        fromBlock=0, argument_filters={"_from": b, "_to": a}
+    )
+    balance_updates = [
+        (
+            (event["blockNumber"], event["logIndex"], event["transactionHash"]),
+            event["args"]["_value"],
+        )
+        for event in balance_update_events
+    ]
+    balance_updates.extend(
+        (
+            (event["blockNumber"], event["logIndex"], event["transactionHash"]),
+            -event["args"]["_value"],
+        )
+        for event in reverse_balance_update_events
+    )
+    balance_updates.sort()
+    return balance_updates
+
+
+def get_transfer_balance_update_events(currency_network_contract, transfer_event):
```

Comment:
Could rename to `get_balance_update_events_for_transfer`.
Could also rename function that return sorted events to precise the fact that they also sort like `get_sorted_interests_for_trustline`, etc ...

Reply:
Renamed, but not the sorted one.